packages/strapi-plugin-content-manager/admin/src/components/RelationPreviewList/index.js

```diff
@@ -0,0 +1,51 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { get } from 'lodash';
+import { Flex, Padded, Count } from '@buffetjs/core';
+import { useIntl } from 'react-intl';
+
+import { getTrad } from '../../utils';
+import { useContentTypeLayout } from '../../hooks';
+import { Truncate, Truncated } from '../CustomTable/styledComponents';
+import CountWrapper from './CountWrapper';
+
+const RelationPreviewList = ({ name, value }) => {
+  const { formatMessage } = useIntl();
+  const { contentType } = useContentTypeLayout();
```

Comment:
Here I suggest doing the following update:

In the `useFetchContentTypeLayout/utils/formatLayout` hook you can add the mainField to the list layout:

```

const formatListLayoutWithMetas = obj => {
  const formatted = obj.layouts.list.reduce((acc, current) => {
    const fieldSchema = get(obj, ['attributes', current], {});
    let metadatas = get(obj, ['metadatas', current, 'list'], {});

    const type = fieldSchema.type;

    if (type === 'relation') {
      metadatas = {
        ...metadatas,
        mainField: get(obj, ['metadatas', current, 'edit', 'mainField'], 'id'),
      };
    }

    acc.push({ key: `__${current}_key__`, name: current, fieldSchema, metadatas });

    return acc;
  }, []);

  return formatted;
};
```

And then in the `List/reducer` you will need to perform the same logic:

```
case ON_CHANGE_LIST_HEADERS: {
  const {
    target: { name, value },
  } = action;

  if (!value) {
    const { metadatas, attributes } = state.contentType;
    let metas = metadatas[name].list;

    if (attributes[name].type === 'relation') {
      const mainField = metadatas[name].edit.mainField;

      metas = { ...metas, mainField };
    }

    drafState.displayedHeaders.push({
      name,
      fieldSchema: attributes[name],
      metadatas: metas,
      key: `__${name}_key__`,
    });
  } else {
    drafState.displayedHeaders = state.displayedHeaders.filter(
      header => header.name !== name
    );
  }
  break;
}
```


With this update you can remove the dependency to the useContentTypeLayout hook

`const RelationPreviewList = ({ metadatas: { mainField }, value)  => {}`